src/mono/wasm/runtime/memory.ts

```diff
@@ -41,33 +45,86 @@ export function _release_temp_frame(): void {
 type _MemOffset = number | VoidPtr | NativePointer | ManagedPointer;
 type _NumberOrPointer = number | VoidPtr | NativePointer | ManagedPointer;
 
+function is_int_in_range(value: Number, min: Number, max: Number) {
+    mono_assert(typeof value === "number", () => `Value is not integer but ${typeof value}`);
+    mono_assert(Number.isInteger(value), "Value is not integer but float");
```

Comment:
We can do ```Number.isSafeInteger(value)``` to handle the first two checks in one go, and for i52 we don't need to do additional range checks I think

Reply:
Yes, I did it to differentiate exception message. I'm ok to make it faster.